File: medplum-java/medplum-java-model/src/main/java/com/fhir/SubstanceSpecification_Relationship.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.Boolean;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableSubstanceSpecification_Relationship.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface SubstanceSpecification_Relationship {
  Optional<Reference> substanceReference();

  Optional<Quantity> amountQuantity();

  Optional<CodeableConcept> amountType();

  Optional<List<Extension>> modifierExtension();

  Optional<Ratio> amountRatio();

  Optional<String> id();

  Optional<CodeableConcept> relationship();

  Optional<List<Reference>> source();

  Optional<String> amountString();

  Optional<List<Extension>> extension();

  Optional<CodeableConcept> substanceCodeableConcept();

  Optional<Ratio> amountRatioLowLimit();

  Optional<Boolean> isDefining();

  Optional<Range> amountRange();

  static ImmutableSubstanceSpecification_Relationship.Builder builder() {
    return ImmutableSubstanceSpecification_Relationship.builder();
  }
}

File: medplum-typescript/src/model/fhir/SubstanceSpecification_Relationship.ts
import { CodeableConcept } from "./CodeableConcept";
import { Extension } from "./Extension";
import { Quantity } from "./Quantity";
import { Range } from "./Range";
import { Ratio } from "./Ratio";
import { Reference } from "./Reference";

export interface SubstanceSpecification_Relationship {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    substanceReference: Reference | null | undefined;
    substanceCodeableConcept: CodeableConcept | null | undefined;
    relationship: CodeableConcept | null | undefined;
    isDefining: boolean | null | undefined;
    amountQuantity: Quantity | null | undefined;
    amountRange: Range | null | undefined;
    amountRatio: Ratio | null | undefined;
    amountString: string | null | undefined;
    amountRatioLowLimit: Ratio | null | undefined;
    amountType: CodeableConcept | null | undefined;
    source: Reference[] | null | undefined;
}
